st2tests/st2tests/fixtures/packs/dummy_pack_11/config.schema.yaml

```diff
@@ -0,0 +1,5 @@
+---
+  api_key:
+    type: "string"
+    secret: true
+    required: true
```

Comment:
Can you add a separate test case where `default` is used in the config schema and `decrypt_kv` is in the default value?

Reply:
What you are asking for here is validation of the schema itself. I am not sure if we validate the schema itself. If we do, I would think that schema validation should happen during pack installation/registration. I am fairly confident that we don't validate the schema as I tested by giving `type: "array"` and `default: 8000` and it did not complain. 

Let me know if my thinking is not correct and if there is another place I need to look at. We should track this request as another issue as this is for validating the config schema and not the config itself. This PR is for validating the config.

Reply:
I'm not asking to do JSON schema validation here.  In the `validate_config_against_schema` method, when the util_schema does a validate on the config instance, it will also assign default values at https://github.com/StackStorm/st2/blob/9edc3fb6da7f12515f6011960f1b72538e53368e/st2common/st2common/util/schema/__init__.py#L301.  So what if you move this decrypt_kv check until after the validate at https://github.com/StackStorm/st2/blob/master/st2common/st2common/util/pack.py#L110?  Will this catch the decrypt_kv in default value?

Reply:
Done